orchestra/utils/task_lifecycle.py

```diff
@@ -275,6 +228,182 @@ def assign_task(worker_id, task_id):
+
+
+@transaction.atomic
+def _revert_assignment_to_datetime(assignment, revert_datetime, fake=False):
+    audit = None
+    # Revert assignment to datetime by removing snapshots
+    gt_idx = None
+    snapshots = assignment.snapshots['snapshots']
+    for i, snapshot in enumerate(snapshots):
+        if parse_datetime(snapshot['datetime']) > revert_datetime:
+            gt_idx = i
+            break
+    if gt_idx is not None:
+        # Revert to before this snapshot
+        audit = {
+            'assignment': assignment,
+            'change': 'reverted',
+            'snapshots': [
+                {
+                    'index': snapshots.index(snapshot),
+                    'snapshot': snapshot,
+                    'change': 'deleted'
+                }
+                for snapshot in snapshots[gt_idx:]
+            ]
+        }
+        if not fake:
+            assignment.snapshots['snapshots'] = snapshots[:gt_idx]
+            assignment.save()
+    return audit
+
+
+@transaction.atomic
+def _revert_task_status(task, fake=False):
+    assignments = assignment_history(task)
+    num_assignments = assignments.count()
+
+    reverted_status = None
+    current_assignment_counter = None
+    previous_assignment_counter = None
+    if num_assignments == 0:
+        # No assignment is present
+        reverted_status = Task.Status.AWAITING_PROCESSING
+    else:
+        last_active_assignment = _get_last_active_assignment(task)
+        if not last_active_assignment:
+            # Task has an assignment but no snapshots
+            reverted_status = Task.Status.PROCESSING
+            current_assignment_counter = 0
+        else:
+            latest_counter = last_active_assignment.assignment_counter
+            snapshot = last_active_assignment.snapshots['snapshots'][-1]
+            if snapshot['type'] == TaskAssignment.SnapshotType.REJECT:
+                # Task was last rejected back to previous worker
+                reverted_status = Task.Status.POST_REVIEW_PROCESSING
+                current_assignment_counter = latest_counter - 1
+                previous_assignment_counter = latest_counter
+
+            elif (snapshot['type'] == TaskAssignment.SnapshotType.SUBMIT
+                  or snapshot['type'] == TaskAssignment.SnapshotType.ACCEPT):
+                if latest_counter == num_assignments - 1:
+                    # Task was last submitted and no higher-level
+                    # assignments are present (reverted tasks will never end
+                    # in a completed state)
+                    reverted_status = Task.Status.PENDING_REVIEW
+                else:
+                    reverted_status = Task.Status.REVIEWING
+                    previous_assignment_counter = latest_counter
+                    current_assignment_counter = latest_counter + 1
+
+    if reverted_status is None:
+        raise TaskStatusError('Task is malformed; cannot revert status.')
+
+    if current_assignment_counter:
```

Comment:
if current_assignment_counter is 0, this will not fire---are you sure you don't want to explicitly test for None